contracts/modules/Checkpoint/ScheduledCheckpoint.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity ^0.4.24;
+
+import "./ICheckpoint.sol";
+import "../TransferManager/ITransferManager.sol";
+import "../../interfaces/ISecurityToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Burn module for burning tokens and keeping track of burnt amounts
+ */
+contract ScheduledCheckpoint is ICheckpoint, ITransferManager {
+    using SafeMath for uint256;
+
+    struct Schedule {
+        bytes32 name;
+        uint256 startTime;
+        uint256 nextTime;
+        uint256 interval;
+        uint256 index;
+        uint256[] checkpointIds;
+        uint256[] timestamps;
+        uint256[] periods;
+    }
+
+    bytes32[] public names;
+
+    mapping (bytes32 => Schedule) schedules;
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress) public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+     * @notice This function returns the signature of configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+    /**
+     * @notice adds a new schedule for checkpoints
+     * @param _name name of the new schedule (must be unused)
+     * @param _startTime start time of the schedule (first checkpoint)
+     * @param _interval interval at which checkpoints should be created
+     */
+    function addSchedule(bytes32 _name, uint256 _startTime, uint256 _interval) external onlyOwner {
+        require(_startTime > now, "Start time must be in the future");
+        require(schedules[_name].name == bytes32(0), "Name already in use");
+        schedules[_name].name = _name;
+        schedules[_name].startTime = _startTime;
+        schedules[_name].nextTime = _startTime;
+        schedules[_name].interval = _interval;
+        schedules[_name].index = names.length;
+        names.push(_name);
+    }
+
+    /**
+     * @notice removes a schedule for checkpoints
+     * @param _name name of the schedule to be removed
+     */
+    function removeSchedule(bytes32 _name) external onlyOwner {
+        require(schedules[_name].name == _name, "Name does not exist");
+        uint256 index = schedules[_name].index;
+        names[index] = names[names.length - 1];
+        names.length--;
+        if (index != names.length) {
+            schedules[names[index]].index = index;
+        }
+        delete schedules[_name];
+    }
```

Comment:
I think it is not worth to delete the data from the mapping because there is no another way to access the data once it is deleted from the contract (besides accessing the previous blocks). we can add a boolean flag to make it inactive.
I think dApp may need to show the old data as well if that schedule is not active.

Reply:
Make sense.